source/Mlos.Python/mlos/Spaces/HypergridAdapters/CategoricalToOneHotEncodedHypergridAdapter.py

```diff
@@ -0,0 +1,268 @@
+
+    """
+
+    def __init__(
+            self,
+            adaptee: Hypergrid,
+            merge_all_categorical_dimensions: bool = False,
+            drop: str = None
+    ):
+        if not HypergridAdapter.is_like_simple_hypergrid(adaptee):
+            raise ValueError("Adaptee must implement a Hypergrid Interface.")
+
+        HypergridAdapter.__init__(self, name=adaptee.name, random_state=adaptee.random_state)
+
+        self._adaptee: Hypergrid = adaptee
+        self._merge_all_categorical_dimensions = merge_all_categorical_dimensions
+        self._one_hot_encoder_kwargs = {
+            'drop': drop,
+            'dtype': np.float64,
+            'sparse': False,
+            'handle_unknown': 'error'
+        }
+        self._all_one_hot_encoded_target_dimension_names = []
+        self._adaptee_to_target_data_dict: Dict[str, CategoricalToOneHotEncodingAdapteeTargetMapping] = {}
+        self._adaptee_expected_dimension_name_ordering = []
+        self._concatenation_delim = '___'
+        self._merged_categorical_dimension_column_name = 'ohe_cross_product'
+        self.ohe_target_column_suffix = '__ohe'
+        self._target: Hypergrid = None
+        self.has_adaptee_been_flattened = False
+
+        # Since CategoricalDimension values may have different types within the same dimension,
+        #  we pass the adaptee through the CategoricalToDiscrete adapter to move all value types to ints
+
+        # Because the OneHotEncoder needs to remember the dimension names (which change by the flattening in CategoricalToDiscrete),
+        #  the flattening is performed here so the OneHotEncoder discovers the correct flattened column names
+        if self._adaptee.is_hierarchical():
+            self._adaptee = HierarchicalToFlatHypergridAdapter(adaptee=self._adaptee)
+            self.has_adaptee_been_flattened = True
+
+        # Since the CategoricalToDiscrete adapter converts categorical dimensions to discrete dimensions, we remember the categorical dim names
+        self._adaptee_dimension_names_to_transform = []
+        for adaptee_dimension in self._adaptee.dimensions:
+            if isinstance(adaptee_dimension, CategoricalDimension):
+                self._adaptee_dimension_names_to_transform.append(adaptee_dimension.name)
+            self._adaptee_expected_dimension_name_ordering.append(adaptee_dimension.name)
+
+        if any(isinstance(dimension, CategoricalDimension) for dimension in self._adaptee.dimensions) or self.has_adaptee_been_flattened:
+            self._adaptee = CategoricalToDiscreteHypergridAdapter(adaptee=self._adaptee)
+
+        self._build_simple_hypergrid_target()
+
+    @property
+    def adaptee(self) -> Hypergrid:
+        return self._adaptee
+
+    @property
+    def target(self) -> Hypergrid:
+        return self._target
+
+    def get_original_categorical_column_names(self):
+        return self._adaptee_to_target_data_dict.keys()

```

Comment:
Could see also providing access to the the map for lookup instead of just the array of names, but that could always be added later when the need arises.